basex-core/src/main/java/org/basex/core/locks/DBLocking.java

```diff
@@ -138,7 +138,9 @@ public void acquire(final Proc pr, final StringList read, final StringList write
       }
       // global read locking
       if(read == null) {
-        while(localWriters > 0) {
+        while(localWriters > 0 &&
+            // We're the only writer, allow global read lock anyway
+            !(1 == localWriters && !(null == write || write.isEmpty()))) {
```

Comment:
Yes it could be, but I decided to skip the "normalization" and stick closer to the semantics: "There's only one writer, and we're this one".